findbugs-test-util/src/test/java/com/h3xstream/findbugs/test/service/ClassFileLocator.java

```diff
@@ -63,12 +69,18 @@ public String getJarFilePath(String path) {
 
     private String getFilenameFromUrl(URL url) {
         String filename;
-		try {
-			filename = url.toURI().getPath();
-		} catch (final URISyntaxException e) {
-			fail("Failed to get file path = " + url, e);
-			return null;
-		}
+        try {
+            if (TEST_JAR_MATCHER.matcher(url.getPath()).find()) {
```

Comment:
Attempt to replace in-jar file path with classpath.  Original url returned null.  Not sure if this approach is the best solution or if there is another mechanism to locate classes within dependency jars.

Reply:
I barely remember writing this code. 🤣

It seems generic enough for a utility class. (some other spotbugs plugins are using this)